### ISSUE TITLE
[FEATURE] Changing Variable values using UnityEvents is not possible

### ISSUE THREAD

**Issue by sampaiodias** [AdamRamberg/unity-atoms#82]
**Is your feature request related to a problem? Please describe.**
Designers can't properly change the value of AtomVariables without scripts because AtomVariables don't have methods that support this. For example, it is not possible to increment an IntVariable using a UnityEvent because ApplyChange(int amount) returns bool instead of void.

**Describe the solution you'd like**
We need a method like ApplyChange but with void return. It can either be the same method or a multitude of methods, like Set and Add.

```csharp
using UnityEngine;

namespace UnityAtoms
{
    /// <summary>
    /// Variable of type `int`. Inherits from `EquatableAtomVariable&lt;int, IntEvent, IntIntEvent&gt;`.
    /// </summary>
    [EditorIcon("atom-icon-lush")]
    [CreateAssetMenu(menuName = "Unity Atoms/Variables/Int", fileName = "IntVariable")]
    public sealed class IntVariable : EquatableAtomVariable<int, IntEvent, IntIntEvent>, IWithApplyChange<int, IntEvent, IntIntEvent>
    {
        public bool ApplyChange(int amount)
        {
            return SetValue(Value + amount);
        }

        public bool ApplyChange(EquatableAtomVariable<int, IntEvent, IntIntEvent> amount)
        {
            return SetValue(Value + amount.Value);
        }

        public void Set(int amount)
        {
            SetValue(amount);
        }

        public void Add(int amount)
        {
            SetValue(Value + amount);
        }
    }
}
```

**Describe alternatives you've considered**
Some kind of script or asset that implements these methods instead, but it seems very odd and unintuitive.

**Additional context**
Here's a 'bad' workaround that solves this:

```csharp
using UnityEngine;

namespace UnityAtoms.Examples
{
    [AddComponentMenu("Unity Atoms/Examples/Int Setter")]
    public sealed class IntSetter : MonoBehaviour
    {
        [SerializeField]
        private IntVariable _variable;

        public void Set(int amount)
        {
            _variable.Value = amount;
        }

        public void Add(int amount)
        {
            _variable.Value += amount;
        }

        public void Subtract(int amount)
        {
            _variable.Value -= amount;
        }
    }
}
```

**Comment by AdamRamberg** [comment]
Thanks for submitting this issue! 

Is this not already possible? Not applying (adding or subtracting) a value, but setting a value. Seems to work for me when setting a Test variable using the `Value` property in the demo scene: 
![Skärmavbild 2019-11-11 kl  08 00 45](https://user-images.githubusercontent.com/10260896/68567531-7cc35d00-0459-11ea-856e-c8626a98716d.png)

The second solution is also already supported using the `SetVariableValue` Atom Actions. 

Guess the only case not covered is `ApplyChange`. Wouldn't mind adding an `ApplyChange` that returns void that just calls the current version of `ApplyChange`.

**Comment by sampaiodias** [comment]
Seems right, ApplyChange (to add, subtract, etc.) is the one missing for UnityEvents

**Comment by AdamRamberg** [comment]
Wouldn't mind adding those. I propose to add the following methods then to `IntVariable` and `FloatVariable`: 
- `AddTo` - Add value to Variable value
- `SubtractFrom` - Subtract value from Variable value
- `MultiplyBy` - Multiply Variable value by value
- `DivideBy` - Divide Variable value by value

**Comment by AdamRamberg** [comment]
If anyone wants to contribute to Unity Atoms feel free to grab this issue and create a PR. Would appreciate if you assign yourself to this issue if you do. <3

**Comment by sampaiodias** [comment]
After thinking about it more, here are my list of method suggestions:

IntVariable
- public void Add(int value)
- public void Subtract(int value)
- public void MultiplyBy(int value)
- public void DivideBy(int value)

FloatVariable
- public void Add(float value)
- public void Subtract(float value)
- public void MultiplyBy(float value)
- public void DivideBy(float value)

Vector2Variable
- public void MultiplyBy(float value)
- public void DivideBy(float value)

Vector3Variable
- public void MultiplyBy(float value)
- public void DivideBy(float value)

ColorVariable
- public void SetAlpha(float value)

StringVariable
- public void Prepend(string text) //Adds a string at the start of the original string
- public void Append(string text) //Adds a string at the end of the original string (classic concatenation)

**Comment by soraphis** [comment]
Modulo? "DivideByToModulo" or "DivideBy(To)Remainder"

**Comment by sampaiodias** [comment]
Not sure if I'm getting it wrong, but I don't particularly see people using these haha

**Comment by sampaiodias** [comment]
Doesn't hurt to have them, though

**Comment by soraphis** [comment]
> Not sure if I'm getting it wrong, but I don't particularly see people using these haha

I don't get this whole feature. at all. but: why should other basic operations be more valuable?

example, you want every third time to happen something interesting. you do a (value +1)%3 add an listener to it that listens to the variable becoming 0 and does then something fancy

**Comment by sampaiodias** [comment]
I get it now, thanks! This feature of having additional methods is to allow people to easily change values of Variables (not just hard setting the value) using UnityEvents without the need to create a script just to do that. A simple add operation shouldn't require someone to create a whole script just for that.

**Comment by AdamRamberg** [comment]
Implemented in #115
